pkg/config/user_config.go

```diff
@@ -291,16 +291,54 @@ type KeybindingSubmodulesConfig struct {
 
 // OSConfig contains config on the level of the os
 type OSConfig struct {
-	// EditCommand is the command for editing a file
+	// Command for editing a file. Should contain "{{filename}}".
+	Edit string `yaml:"edit,omitempty"`
+
+	// Command for editing a file at a given line number. Should contain
+	// "{{filename}}", and may optionally contain "{{line}}".
+	EditAtLine string `yaml:"editAtLine,omitempty"`
+
+	// Same as EditAtLine, except that the command needs to wait until the
+	// window is closed.
+	EditAtLineAndWait string `yaml:"editAtLineAndWait,omitempty"`
+
+	// Whether the given edit commands use the terminal. Used to decide whether
+	// lazygit needs to suspend to the background before calling the editor.
+	// Pointer to bool so that we can distinguish unset (nil) from false.
+	EditInTerminal *bool `yaml:"editInTerminal,omitempty"`
+
+	// A built-in preset that sets all of the above settings. Supported presets
+	// are "vim", "emacs", "nano", "vscode", "sublime", "bbedit", "xcode".
```

Comment:
I wouldn't enumerate the presets here given they'll change over time and it'll be one more place we need to keep in sync. I'd instead reference the place where the presets are defined

Reply:
Makes sense, see ed17d96e023c.